graphql-server/pkg/queries/proposal.go

```diff
@@ -121,3 +122,12 @@ func (q *ProposalQuery) QueryProposalTallyResults(ids []int) ([]*models.Proposal
 
 	return result, nil
 }
+
+func (q *ProposalQuery) QueryProposalByIDs(ids []string) ([]*models.Proposal, error) {
+	proposals := make([]*models.Proposal, len(ids))
+	err := q.NewQuery().Where("id IN (?)", bun.In(ids)).Scan(q.ctx, &proposals)
```

Comment:
I don't know if `bun` has been improved. Please check if error will be thrown when `len(ids) == 0`

Reply:
Yes it will throw an error

```bash
time="2022-06-10T11:48:49+08:00" level=error msg="Unhandled internal error: ERROR: syntax error at or near \")\" (SQLSTATE=42601)" error="ERROR: syntax error at or near \")\" (SQLSTATE=42601)" path=proposalByIDs
```

SQL stmt causing this (copied from logs):

```sql
SELECT "proposal"."id", "proposal"."title", "proposal"."description", "proposal"."proposal_route", "proposal"."proposal_type", "proposal"."submit_time", "proposal"."deposit_end_time", "proposal"."voting_start_time", "proposal"."voting_end_time", "proposal"."proposer_address", "proposal"."status" 
FROM "proposal" WHERE (id IN ())
```

should I add a check for `len(ids) == 0` before querying the database, and return empty list if true?

```go
if len(ids) == 0 {
  nil, nil
}
```

Reply:
I think we should return empty list to match the dataloader convention